src/Collateral.hs

```diff
@@ -100,31 +101,28 @@ mkValidator contractInfo@ContractInfo{..} dat rdm ctx = validate
     validateDebtAndInterestAmnt :: Bool
     validateDebtAndInterestAmnt = not ((interestcs dat == loancs dat) && (interesttn dat == loantn dat)) || (getLoanAmnt (U.valueToSc interestscvh ctx) >= loanamnt dat + interestamnt dat)
 
-    lenderNftFilter :: (CurrencySymbol, TokenName, Integer) -> Bool
-    lenderNftFilter (cs, _tn, n) = cs == lenderNftCs && n == 1
-
-    filterValues :: Value -> [(CurrencySymbol, TokenName, Integer)]
-    filterValues v = filter lenderNftFilter $ flattenValue v
+    validateBorrowerNftBurn :: Bool
+    validateBorrowerNftBurn = any (\(cs, tn, n) -> cs == borrowersNFT dat && tn == borrower && n == (-1)) (U.mintFlattened ctx)
 
-    containsFlattenedValue :: [(CurrencySymbol, TokenName, Integer)] -> (CurrencySymbol, TokenName, Integer) -> Bool
-    containsFlattenedValue xs (cs, tn, _) = foldr (\(cs', tn', _) acc -> (cs' == cs && tn' == tn) || acc) False xs
+    getCollateralScHashes :: [DatumHash]
+    getCollateralScHashes = map fst (scriptOutputsAt interestscvh (U.info ctx))
```

Comment:
I think the naming is misleading. It actually returns the datum hashes located on the address of the Interest script.
```suggestion
    getInterestScHashes :: [DatumHash]
    getInterestScHashes = map fst (scriptOutputsAt interestscvh (U.info ctx))
```